src/composition.rs

```diff
@@ -439,6 +498,16 @@ impl Composition {
             &mut visitor,
             flux::ast::walk::Node::File(&self.file),
         );
+
+        let mut analyzer = CompositionQueryAnalyzer {
```

Comment:
Ah, yeah, I went back and forth on this idea, and ultimately decided against it. I think the AST has to be the source of state, which is why we visit to get that state out, but I don't think we should drop that AST and generate a new one. The kinda glaring issue with this one is that we don't preserve the order of filters, and we've been pretty clear that that order is important. I think, for now, this approach is kind of a non-starter.